src/Introspection.cpp

```diff
@@ -2214,6 +2216,13 @@ bool dump_stack_frame() {
 std::string get_variable_name(const void *var, const std::string &expected_type) {
     if (!debug_sections) return "";
     if (!debug_sections->working) return "";
+
+    // At least one version of libc++ will report data races inside the regex
+    // implementation if we call this from multiple threads at the same time
+    // (e.g. from correctness_vector_width); this mutex serves more to silence
+    // what is likely to be an irrelevant-to-us data race when using TSAN.
+    std::lock_guard<std::mutex> lock(get_variable_name_lock);
```

Comment:
This entirely serializes something potentially expensive (stack walking code). I don't think it's a good solution.